services/productService.js

```diff
@@ -0,0 +1,29 @@
+const productDao = require("../models/productDao.js");
+
+const allProducts = async (page, pageSize, category, sub, type, sort) => {
+  let start = 0;
+  if (page <= 0) {
+    page = 1;
+  } else {
+    start = (page - 1) * pageSize;
+  }
+  const moreThanProducts = await productDao.moreThanProducts();
+  if (page > Math.ceil(moreThanProducts[0]["count(*)"] / pageSize)) {
+    const err = new Error("제품이 없습니다.");
+    err.statusCode = 400;
+    throw err;
+  }
+  const getProduct = await productDao.getProduct(
+    start,
+    pageSize,
+    category,
+    sub,
+    type,
+    sort
+  );
+  return getProduct;
+};
```

Comment:
마이너하지만 변수명 수정해주면 좋을 것 같습니다~!
결과 자체만 보면 products 리스트가 반환되기 때문입니다.
```suggestion
  // 혹은 products 
  const result = await productDao.getProduct(
    start,
    pageSize,
    category,
    sub,
    type,
    sort
  );
  return result;
};
```